doc/modules/neighbors.rst

```diff
@@ -464,6 +464,39 @@ leaf nodes.  The level of this switch can be specified with the parameter
 
 ``leaf_size`` is not referenced for brute force queries.
 
+Valid Metrics for Nearest Neighbor Algorithms
+---------------------------------------------
+
+========================     =================================================================
+Algorithm                    Valid Metrics
+========================     =================================================================
+**Brute Force**              'euclidean', 'l2', 'l1', 'manhattan', 'cityblock',
+                             'braycurtis', 'canberra', 'chebyshev', 'correlation',
+                             'cosine', 'dice', 'hamming', 'jaccard', 'kulsinski',
+                             'mahalanobis', 'matching', 'minkowski', 'rogerstanimoto',
+                             'russellrao', 'seuclidean', 'sokalmichener',
+                             'sokalsneath', 'sqeuclidean', 'yule', 'wminkowski'
+
+**K-D Tree**                 'chebyshev', 'euclidean', 'cityblock', 'manhattan', 'infinity',
+                             'minkowski', 'p', 'l2', 'l1'
+
+**Ball Tree**                'chebyshev', 'sokalmichener', 'canberra', 'haversine',
+                             'rogerstanimoto', 'matching', 'dice', 'euclidean', 'braycurtis',
+                             'russellrao', 'cityblock', 'manhattan', 'infinity', 'jaccard',
+                             'seuclidean', 'sokalsneath', 'kulsinski', 'minkowski',
+                             'mahalanobis', 'p', 'l2', 'hamming', 'l1', 'wminkowski', 'pyfunc'
+========================     =================================================================
```

Comment:
[As @jnothman suggested](https://github.com/scikit-learn/scikit-learn/pull/4525#issuecomment-308404236), we might prefer referring to the table present in `modules/generated/sklearn.neighbors.DistanceMetric.html` to avoid duplication.